Src/GUI/Widgets/OutputFilesWidget.py

```diff
@@ -1,29 +1,52 @@
-# **************************************************************************
-#
-# MDANSE: Molecular Dynamics Analysis for Neutron Scattering Experiments
-#
-# @file      Src/GUI/Widgets/OutputFilesWidget.py
-# @brief     Implements module/class/test OutputFilesWidget
-#
-# @homepage  https://mdanse.org
-# @license   GNU General Public License v3 or higher (see LICENSE)
-# @copyright Institut Laue Langevin 2013-now
-# @authors   Scientific Computing Group at ILL (see AUTHORS)
-#
-# **************************************************************************
-
+# **************************************************************************
+#
+# MDANSE: Molecular Dynamics Analysis for Neutron Scattering Experiments
+#
+# @file      Src/GUI/Widgets/OutputFilesWidget.py
+# @brief     Implements module/class/test OutputFilesWidget
+#
+# @homepage  https://mdanse.org
+# @license   GNU General Public License v3 or higher (see LICENSE)
+# @copyright Institut Laue Langevin 2013-now
+# @authors   Scientific Computing Group at ILL (see AUTHORS)
+#
+# **************************************************************************
+
 import os
+import os.path
 
 import wx.combo
 import wx.lib.filebrowsebutton as wxfile
 
 from MDANSE import REGISTRY
 
+from MDANSE.GUI import PUBLISHER
 from MDANSE.GUI.ComboWidgets.ComboCheckbox import ComboCheckbox
 from MDANSE.GUI.Widgets.IWidget import IWidget
 
+def get_unique_filename(directory,basename):

```

Comment:
This is not large, but I was wondering if this function should not be a method of the class instead, given that it is solely related to operations in the class. What do you think?